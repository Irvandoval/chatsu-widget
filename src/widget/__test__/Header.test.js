import React from 'react';
import Header from '../Header';
import { JSDOM } from 'jsdom';
import { shallow } from 'enzyme';

const onClick = jest.fn();

const conf = {
  closedStyle: "chat",
  desktopHeight: 450,
  desktopWidth: 370,
  displayMessageTime: true
};

describe('<Header />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Header onClick={onClick} conf={conf}/>);
    expect(tree).toMatchSnapshot();
  });
})