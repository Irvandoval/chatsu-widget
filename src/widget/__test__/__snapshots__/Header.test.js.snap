// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches the snapshot 1`] = `
<div
  onClick={[MockFunction]}
  style={
    Object {
      "cursor": "pointer",
      "position": "relative",
    }
  }
>
  <div
    className="desktop-closed-message"
    style={
      Object {
        "background": undefined,
        "borderRadius": "5px",
        "boxShadow": "#8e8d8d -3px 2px 20px",
        "color": "#fff",
        "display": "block",
        "fontSize": "14px",
        "letterSpacing": "1px",
        "marginTop": "23px",
        "padding": "15px 20px",
        "position": "absolute",
        "right": "130px",
        "top": 0,
      }
    }
  >
    <div
      style={
        Object {
          "borderBottom": "10px solid transparent",
          "borderLeft": "10px solid undefined",
          "borderTop": "10px solid transparent",
          "height": 0,
          "position": "absolute",
          "right": "-10px",
          "top": "12px",
          "width": 0,
        }
      }
    />
  </div>
  <div
    className="desktop-closed-message-avatar"
    style={
      Object {
        "background": "#fff",
        "border": "2px solid undefined",
        "borderRadius": "999px",
        "boxShadow": "#8e8d8d -3px 2px 20px",
        "display": "block",
        "height": "60px",
        "position": "absolute",
        "right": "40px",
        "top": "10px",
        "width": "60px",
      }
    }
  >
    <img
      alt="Avatar"
      style={
        Object {
          "borderRadius": "999px",
          "height": "auto",
          "width": "100%",
        }
      }
    />
    <div
      style={
        Object {
          "background": "#d0021b",
          "borderRadius": "999px",
          "bottom": "-5px",
          "color": "#fff",
          "fontSize": "0.8em",
          "fontWeight": 600,
          "height": "20px",
          "lineHeight": "20px",
          "position": "absolute",
          "right": "-5px",
          "textAlign": "center",
          "width": "20px",
        }
      }
    >
      1
    </div>
  </div>
</div>
`;
